### PR TITLE
Use compatible `MessageAckBuilder` to instead of `AckHandler`

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

In the current implementation, we use `AckHandler` to process all things about message ack. the process is:

1. Create a high abstract `AckBuilder` with a `success` flag and other parameters.
2. `AckHandler` determines how to build a corresponding MQTT message based on `success` and uses another compatible Builder to build it.
3. `AckHandler` chose to `send` message or `sendThenClose`.

This process is complex and difficult to understand. I think we can easily remove this part of the model and use the `compatible Builder` mentioned in step 2 directly.

### Modifications

- Remove `AckHandler` model and relative logic.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

**Comment by mattisonchao** [comment]
Hi, @Technoboy- 
This PR passed all tests, but we still have some `Codacy` check errors (we already know) that include:
- Switch statements should be exhaustive, add a default case (or missing enum branches) [Decoder]
- The utility class name 'MqttPubAck' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'  [AckBuilder]

PTAL :-), when you have tiem.

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java [hunk 1]
```diff
@@ -123,61 +121,52 @@ private void addIdleStateHandler() {
...
+        return new MqttAdapterMessage(clientId, mqttMessage, !isFromProxy());
     }
 
     /**
-     * Broker send disconnect.
+     * Disconnect the current connection and send a disconnect MqttMessage if needed.
      */
     public void disconnect() {
-        if (MqttUtils.isMqtt5(protocolVersion) || isAdapter()) {
+        if (this.isFromProxy() || !MqttUtils.isMqtt3(protocolVersion)) {
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java:158]: !MqttUtils.isMqtt3 -> MqttUtils.isNotMqtt3

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java [hunk 2]
```diff
@@ -123,61 +121,52 @@ private void addIdleStateHandler() {
...
-
-    public ChannelFuture send(MqttAdapterMessage adapterMessage) {
-        adapterMessage.setAdapter(isAdapter());
+    /**
+     * Convert #{MqttMessage} to #{MqttAdapterMessage} and encode type set by connection #{fromProxy} flag.
+     * Then send the message.
+     * @param mqttMessage Mqtt message
+     * @return CompletableFuture
+     */
+    public CompletableFuture<Void> convertAndSend(MqttMessage mqttMessage) {
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java:130]: convertAndSend -> send

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java [hunk 3]
```diff
@@ -278,9 +243,31 @@ public int getServerReceivePubMessage() {
         return SERVER_CURRENT_RECEIVE_PUB_MAXIMUM_UPDATER.get(this);
     }
 
-    /**
-     * Connection state.
-     */
+    public void updateStateAndSendConnAck() {
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java:246]: updateStateAndSendConnAck -> SendConnAck

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            } else {
+                return Optional.of(MqttMessageBuilders.pubAck()
+                        .reasonCode(reasonCode.byteValue())
+                        .packetId(packetId)
+                        .properties(getStuffedProperties())
+                        .build());
+            }
+        }
+
+        private MqttProperties getStuffedProperties() {
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java:109]: getStuffedProperties -> getProperties

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            return this;
+        }
+
+        public MqttPubAck.MqttPubErrorAckBuilder reasonCode(
+                Mqtt5PubReasonCode reasonCode) {
+            this.reasonCode = reasonCode;
+            return this;
+        }
+
+        public Optional<MqttMessage> buildIfSupport() {
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java:97]: buildIfSupport -> build

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java (hunk 1 above)
Comment (mattisonchao) [reply mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java]: fixed

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java (hunk 2 above)
Comment (mattisonchao) [reply mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java]: fixed

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java (hunk 3 above)
Comment (mattisonchao) [reply mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/Connection.java]: fixed

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java (hunk 4 above)
Comment (mattisonchao) [reply mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java]: fixed

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java (hunk 5 above)
Comment (mattisonchao) [reply mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/messages/ack/MqttPubAck.java]: fixed

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**